### PR TITLE
feat: use security-group module instead of resource

### PR DESCRIPTION
**Pull request by SweetOps**
## what
* use security-group module instead of resource
* update tests

## why
* more flexible  than current implementation 
* bring configuration of security group/rules to one standard

## references
* CPCO-409

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -51,49 +57,17 @@ resource "aws_efs_access_point" "default" {
...
-resource "aws_security_group_rule" "egress" {
-  count             = module.this.enabled ? 1 : 0
-  type              = "egress"
-  from_port         = 0
-  to_port           = 0
-  protocol          = "-1"
-  cidr_blocks       = ["0.0.0.0/0"]
-  security_group_id = join("", aws_security_group.efs.*.id)
+  enabled = local.security_group_enabled
+  name    = format("%s-efs", module.this.id)
```
Comment (osterman) [inline main.tf:69]: hrm... I think `efs` is actually added to attributes

**Review by osterman — DISMISSED** [review]

**Review by nitrocode — APPROVED** [review]

**Comment by nitrocode** [comment]
A list of breaking changes have been added to the release notes.

https://github.com/cloudposse/terraform-aws-efs/releases/tag/0.31.0

**Pull request merged**